### PR TITLE
qmake: add support for AVX2 builds

### PR DESCRIPTION
**Pull request by thiagomacieira**
This is ugly, but works for qt3d.

Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>

### CODE REVIEW

**Comment by bryteise** [comment]
Ugh I don't like this but adding a move script isn't particularly nice either. I guess the script makes more sense though. Perhaps we can add a utility function to autospec for getting the script into the builddir as a tmpfile.

**Comment by thiagomacieira** [comment]
Shall we continue this discussion in the public CLR mailing list for other suggestions?

**Comment by bryteise** [comment]
So I'm inclined to merge this unless somebody really hates the approach. The ideal case is we can maybe move some of the other build patterns to reuse this idea and refine from there I imagine.

**Comment by thiagomacieira** [comment]
Will wait for conclusion on the ML.

File: autospec/specfiles.py [hunk 1]
```diff
@@ -367,6 +367,20 @@ def write_proxy_exports(self):
         self._write_strip("export https_proxy=http://127.0.0.1:9/")
         self._write_strip("export no_proxy=localhost,127.0.0.1,0.0.0.0")
 
+    def write_move_avx2_files(self, libsubdir="haswell", pluginsuffix=".avx2"):
```
Comment (thiagomacieira) [inline autospec/specfiles.py]: I don't understand the error message. The function is clearly defined here.

File: autospec/specfiles.py [hunk 2]
```diff
@@ -367,6 +367,20 @@ def write_proxy_exports(self):
...
+pushd %{{buildroot}}/usr/bin
+for f in *; do mkdir -p {0}; mv -t {0} "$f"; done
+cd ../lib64
+find . -name '*.so*' -print0 | xargs -r0 /usr/bin/clr-avx2-move.pl {0} {1}
+popd
+            """.format(libsubdir, pluginsuffix))
+
+    def write_move_avx512_files(self):
+        """Same as above, but for AVX-512"""
+        write_move_avx2_files(self, "haswell/avx512_1", ".avx512")
```
[the remark this answers is not in the document]
Comment (thiagomacieira) [reply autospec/specfiles.py]: Got it. Fixed.

**Comment by thiagomacieira** [comment]
Will change again, moving `find` into the script or RPM macro.

**Comment by thiagomacieira** [comment]
Updated. The machinery is now hidden in the RPM macros.

**Comment by thiagomacieira** [comment]
Updated to use RPM macros.

**Comment by bryteise** [comment]
So this is good. I'm going to wait to merge this though as I want to let folks systems update to a new enough version to have access to the rpm macro that was just added.

**Comment by thiagomacieira** [comment]
Should be good now?

**Pull request merged**